torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -339,6 +339,7 @@ class NCCLComm {
     // Set true failure reason if provided by ProcessGroupNCCL (e.g. work
     // timeout)
     commFailureReason_ = commFailureReason;
+    LOG(INFO) << "Aborting ncclComm_ with reason: " << *commFailureReason;
```

Comment:
maybe cleaner if we just call logPrefix() here directly, instead of embedding it in the reason.  maybe have to move this function to a .cpp to accomplish that but that should be ok?

also, much nicer if we have specific communicator id here.  Is it bad to log the comm id that is a url/hash or  whatever?  Is it unique? If it's unique, we can also log it when creating it early on, then we can easily map from which PG created in which order, to which destroyed in which order.

otoh i am not sure if this is necessary. In most cases i think there is 1:1 process-group:communicator.  If that's always true, then logPrefix would give enough info here.  But i am not sure its always true.

cc @kwen2501

Reply:
I think it is 1:1 process-group:communicator and comm id is a struct which contains IP address port info, etc. (@kwen2501 can correct me) I guess it's very nccl specific logic, probably not good to move it out. That's why I embedded PG info in the abort reason.

Reply:
In most cases, yes, there is a 1:1 mapping. 
If within a PG a rank performs both collective and P2P, then potentially there isn't, because in today's implementation, there will be an additional comm for that P2P.

Reply:
> comm id is a struct which contains IP address port info

Strictly speaking, that's ncclUniqueID not comm id. If we save ncclUniqueID in ProcessGroupNCCL, then yes, we can use that to describe comm id. But honestly, I am not sure if that's more indicative than just printing the value of the nccl comm -- it is a pointer pointing to the comm object so it must be unique.

Reply:
My recommendation of solving @wconstab 's ask is to remove the `commFailureReason` signature from this abort() function. Each call site of abort() just log whatever reason they have (good or bad) before calling into it.

Reply:
OK, then I maybe just keep the abort reason untouched and only add one line of log before calling into abort.